tests/unit/modules/test_ssh.py

```diff
@@ -93,18 +93,17 @@ def test_replace_auth_key(self):
         comment_line = '# this is a comment \n'
 
         # Write out the authorized key to a temporary file
-        if salt.utils.is_windows():
-            temp_file = tempfile.NamedTemporaryFile(delete=False)
-        else:
-            temp_file = tempfile.NamedTemporaryFile(delete=False, mode='w+')
-
-        # Add comment
-        temp_file.write(comment_line)
-        # Add empty line for #41335
-        temp_file.write(empty_line)
-        temp_file.write('{0} {1} {2} {3}'.format(options, enc, key, email))
+        temp_file = tempfile.NamedTemporaryFile(delete=False, mode='w+')
         temp_file.close()
 
+        with salt.utils.fopen(temp_file.name, 'w') as _fh:
+            # Add comment
+            _fh.write(comment_line)
+            # Add empty line for #41335
+            _fh.write(empty_line)
+            _fh.write('{0} {1} {2} {3}'.format(options, enc, key, email))
+            _fh.close()
```

Comment:
I don't think you need this `close()` since the context handler should do that for you, but I don't think it hurts either.

Reply:
Good catch, I'll remove it.